plugins/Kaleidoscope-LED-ActiveLayerKeys/src/kaleidoscope/plugin/LED-ActiveLayerKeys.cpp

```diff
@@ -0,0 +1,85 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-LED-ActiveLayerKeys -- Light up the LEDs based on the keys specified in the active layer
+ * Copyright (C) 2018  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "kaleidoscope/plugin/LED-ActiveLayerKeys.h"
+
+#include <Arduino.h>  // for pgm_read_byte
+#include <stdint.h>   // for uint8_t
+
+#include "kaleidoscope/KeyAddr.h"               // for KeyAddr
+#include "kaleidoscope/Runtime.h"               // for Runtime
+#include "kaleidoscope/event_handler_result.h"  // for EventHandlerResult, EventHandlerResult::OK
+#include "kaleidoscope/layers.h"                // for Layer
+#include "kaleidoscope/plugin/LEDControl.h"     // for LEDControl
+
+namespace kaleidoscope {
+namespace plugin {
+
+const cRGB *LEDActiveLayerKeysEffect::colormap_;
+
+LEDActiveLayerKeysEffect::TransientLEDMode::TransientLEDMode(
+  const LEDActiveLayerKeysEffect *parent)
+  : parent_(parent) {}
+
+void LEDActiveLayerKeysEffect::setColormap(const cRGB colormap[]) {
+  colormap_ = colormap;
+}
+
+cRGB LEDActiveLayerKeysEffect::TransientLEDMode::getLayerColor(uint8_t layer) {
+  cRGB color;
+
+  if (layer < sizeof(parent_->colormap_)/sizeof(*parent_->colormap_)) {
```

Comment:
This seems to break it for all but the first layer in my sketch 🤔 How should I check if the color map has an entry for a given layer?